pkg/codegen/hcl2/model/type.go

```diff
@@ -37,6 +37,7 @@ type Type interface {
 
 	AssignableFrom(src Type) bool
 	ConversionFrom(src Type) ConversionKind
+	GetAnnotations() []interface{}
```

Comment:
This should probably return something with an API more like that of `context.Context`: a logical map keyed by package-level variables.